src/js/components/dropdown.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright (C) 2018 The Trustees of Indiana University
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import keyCodes from '../utilities/keyCodes';
+
+export default class Dropdown {
+  constructor(element) {
+    this.element = element;
+    this.focusableElements = 'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]';
+
+    this.dropdownAttribute = '[data-dropdown]';
+
+    this.toggleAttribute = '[data-dropdown-toggle]';
+    this.toggleElement = this.element.querySelector(this.toggleAttribute);
+
+    this.menuAttribute = '[data-dropdown-menu]';
+    this.menuElement = this.element.querySelector(this.menuAttribute);
+
+    // Keeps track of the currently active dropdown and helps with focus management
+    this.activeDropdown = null;
+
+    // Bind methods
+    this._handleClick = this._handleClick.bind(this);
+    this._handleKeydown = this._handleKeydown.bind(this);
+
+    this.init();
+  }
+
+  /**
+   * Opens a dropdown menu
+   * @param {HTMLButtonElement} toggle - a button element which toggles the dropdown
+   * @param {HTMLDivElement} menu - a div containing the dropdown menu items
+   */
+  openMenu(toggle, menu) {
+    const toggleButton = toggle || this.toggleElement;
```

Comment:
Related to the previous comment. you can remove these variable reassignments since you can use the instance properties that hold reference to the toggle and menu instead.